app/react/Map/specs/Map.spec.js

```diff
@@ -368,4 +383,23 @@ describe('Map', () => {
       expect(instance.updateMapStyle).toHaveBeenCalled();
     });
   });
+
+  describe('replaceKeysMapStyleJson', () => {
+    it('should set the map tiler key from the settings', async () => {
+      render();
+      const style = {
+        sources: { openmaptiles: { url: 'URL?key={{MAP_TILER_KEY}}' } },
+        glyphs: 'URL?key={{MAP_TILER_KEY}}',
+      };
+      const expectedKey = 'XYZ';
+      settingsAPI.get.mockResolvedValue({ mapTilerKey: expectedKey });
+      await instance.replaceKeysMapStyleJson(style);
+      const stringifyStyle = JSON.stringify(instance.mapStyle);
+      expect(stringifyStyle).toEqual(expect.not.stringContaining('{{MAP_TILER_KEY}}'));
+      expect(instance.mapStyle.getIn(['sources', 'openmaptiles', 'url'])).toEqual(
+        expect.stringContaining(expectedKey)
+      );
+      expect(instance.mapStyle.getIn(['glyphs'])).toEqual(expect.stringContaining(expectedKey));
```

Comment:
These two are the right expects to have!

Reply:
nice 👍